src/wp-includes/blocks.php

```diff
@@ -40,6 +40,221 @@ function unregister_block_type( $name ) {
 	return WP_Block_Type_Registry::get_instance()->unregister( $name );
 }
 
+/**
+ * Removes the block asset's path prefix if provided.
+ *
+ * @since 5.5.0
+ *
+ * @param string $asset_handle_or_path Asset handle or prefixed path.
+ *
+ * @return string Path without the prefix or the original value.
+ */
+function remove_block_asset_path_prefix( $asset_handle_or_path ) {
+	$path_prefix = 'file:';
+	if ( strpos( $asset_handle_or_path, $path_prefix ) !== 0 ) {
+		return $asset_handle_or_path;
+	}
+	return substr(
+		$asset_handle_or_path,
+		strlen( $path_prefix )
+	);
+}
+
+/**
+ * Generates the name for an asset based on the name of the block
+ * and the field name provided.
+ *
+ * @since 5.5.0
+ *
+ * @param string $block_name Name of the block.
+ * @param string $field_name Name of the metadata field.
+ *
+ * @return string Generated asset name for the block's field.
+ */
+function generate_block_asset_handle( $block_name, $field_name ) {
+	$field_mappings = array(
+		'editorScript' => 'editor-script',
+		'script'       => 'script',
+		'editorStyle'  => 'editor-style',
+		'style'        => 'style',
+	);
+	return str_replace( '/', '-', $block_name ) .
+		'-' . $field_mappings[ $field_name ];
+}
+
+/**
+ * Finds a script handle for the selected block metadata field. It detects
+ * when a path to file was provided and finds a corresponding
+ * asset file with details necessary to register the script under
+ * automatically generated handle name. It returns unprocessed script handle
+ * otherwise.
+ *
+ * @since 5.5.0
+ *
+ * @param array  $metadata Block metadata.
+ * @param string $field_name Field name to pick from metadata.
+ *
+ * @return string|boolean Script handle provided directly or created through
```

Comment:
```suggestion
 * @return string|bool Script handle provided directly or created through
```

Should be the short form, see https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#phpdoc-tags.